app/User.php

```diff
@@ -59,7 +60,21 @@ public function printJobs()
         return $this->hasMany('App\PrintJob');
     }
 
+    public function roles() {
+        return $this->belongsToMany(Role::class, 'role_users');
+    }
+
+    public function inRole(array $roleNames, $objectId = null)
+    {
+        return DB::table('role_users')
+                ->join('roles', 'roles.id', '=', 'role_users.role_id')
+                ->where('role_users.user_id', '=', $this->id)
+                ->whereIn('roles.name', $roleNames)
+                ->where('role_users.object_id', '=', $objectId)
+                ->count() > 0;
+    }
+
     public function isAdmin() {
-        return $this->permission == 1;
+        return $this->inRole(['admin']);
     }
```

Comment:
We should deprecate this function.

We should add @deprecated annotation, and use something like:
```trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);```